tests/main/security-apparmor/task.yaml

```diff
@@ -11,8 +11,9 @@ restore: |
 execute: |
     echo "Then an unconfined action should succeed"
     test-snapd-tools.cmd touch /dev/shm/snap.test-snapd-tools.foo
-    expected="(?s)File: '/dev/shm/snap.test-snapd-tools.foo'.*?Size: 0"
-    stat /dev/shm/snap.test-snapd-tools.foo | grep -Pqz "$expected"
+    if [ ! test -f /dev/shm/snap.test-snapd-tools.foo ]; then
+        echo "Expected /dev/shm/snap.test-snapd-tools.foo to exist"
+    fi
```

Comment:
As @chipaca commented, this should be _just this single line:_
```
test -f /dev/shm/snap.test-snapd-tools.foo
```
No if, no echo.

As it is, besides over-verbose, it's not even correct. It will never fail.